src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -2548,6 +2592,41 @@ private IEnumerable<string> IterateCollectionNames(Transaction tx, JsonOperation
             return result;
         }
 
+        public IEnumerable<Document> GetDocumentsByBucketFrom(DocumentsOperationContext context, int bucket, long etag)
+        {
+            var table = new Table(DocsSchema, context.Transaction.InnerTransaction);
+
+            foreach (var result in GetItemsByBucket(context.Allocator, table, DocsSchema.DynamicKeyIndexes[DocsBucketAndEtagSlice], bucket, etag))
+            {
+                yield return TableValueToDocument(context, ref result.Result.Reader);
+            }
+        }
+
+        public static IEnumerable<Table.SeekResult> GetItemsByBucket(ByteStringContext allocator, Table table, 
+            TableSchema.DynamicKeyIndexDef dynamicIndex, int bucket, long etag)
+        {
+            using (GetBucketAndEtagByteString(allocator, bucket, etag, out var buffer))
+            using (Slice.External(allocator, buffer, buffer.Length, out var keySlice))
+            using (Slice.External(allocator, buffer, buffer.Length - sizeof(long), out var prefix))
+            {
+                foreach (var result in table.SeekForwardFromPrefix(dynamicIndex, keySlice, prefix, 0))
+                {
+                    yield return result;
+                }
+            }
+        }
+
+        public static ByteStringContext<ByteStringMemoryCache>.InternalScope GetBucketAndEtagByteString(
+            ByteStringContext allocator, int bucket, long etag,
+            out ByteString buffer)
+        {
+            var scope = allocator.Allocate(sizeof(int) + sizeof(long), out buffer);
+            *(int*)buffer.Ptr = bucket;
```

Comment:
We don't _have_ to do it, but it would be better if we reversed the bytes here as well.
That would make the sorting of the buckets much more reasonable for us down the line.

Reply:
done